buildkite/bazelci.py

```diff
@@ -2986,13 +2986,14 @@ def try_update_last_green_commit():
     client = BuildkiteClient(org=org_slug, pipeline=pipeline_slug)
     build_info = client.get_build_info(build_number)
 
-    # Find any failing steps other than Buildifier and "try update last green".
+    # Find any failing steps other than Buildifier and steps with soft_fail enabled then "try update last green".
     def has_failed(job):
         state = job.get("state")
         # Ignore steps that don't have a state (like "wait").
         return (
             state is not None
             and state != "passed"
+            and not job.get("soft_failed")
```

Comment:
Should we also verify the `exit_status` matches?

Reply:
Checking "soft_failed" does that, if the job failed with exist status not listed in its soft_fail property during creation, then "soft_failed" will be false.

Reply:
Ahha, I see, thanks!